lightning/src/util/persist.rs

```diff
@@ -243,10 +250,886 @@ where
+	fn do_try_into_outpoint(name: &str) -> Result<OutPoint, io::Error> {
+		let mut parts = name.splitn(2, '_');
+		let txid = if let Some(part) = parts.next() {
+			Txid::from_hex(part).map_err(|_| {
+				io::Error::new(io::ErrorKind::InvalidData, "Invalid tx ID in stored key")
+			})?
+		} else {
+			return Err(io::Error::new(
+				io::ErrorKind::InvalidData,
+				"Stored monitor key is not a splittable string",
+			));
+		};
+		let index = if let Some(part) = parts.next() {
+			part.parse().map_err(|_| {
+				io::Error::new(io::ErrorKind::InvalidData, "Invalid tx index in stored key")
+			})?
+		} else {
+			return Err(io::Error::new(
+				io::ErrorKind::InvalidData,
+				"No tx index value found after underscore in stored key",
+			));
+		};
+		Ok(OutPoint { txid, index })
+	}
+}
+
+impl TryFrom<&MonitorName> for OutPoint {
+	type Error = io::Error;
+
+	fn try_from(value: &MonitorName) -> Result<Self, io::Error> {
+		MonitorName::do_try_into_outpoint(&value.0)
+	}
+}
+
+impl From<OutPoint> for MonitorName {
+	fn from(value: OutPoint) -> Self {
+		MonitorName(format!("{}_{}", value.txid.to_hex(), value.index))
+	}
+}
+
+/// A struct representing a name for an update.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct UpdateName(u64, String);
+
+impl UpdateName {
+	/// Constructs an [`UpdateName`], after verifying that an update sequence ID
+	/// can be derived from the given `name`.
+	pub fn new(name: String) -> Result<Self, io::Error> {
+		match name.parse::<u64>() {
+			Ok(u) => Ok(u.into()),
+			Err(_) => Err(io::Error::new(
+				io::ErrorKind::InvalidData,
+				"cannot parse u64 from update name",
+			)),
+		}
+	}
+
+	/// Convert this monitor update name to a &str
+	pub fn as_str(&self) -> &str {
+		&self.1
+	}
+
+	/// Left-pad the sequential update id for string representation
+	fn left_padded_string(value: u64) -> String {
+		format!("{:0>20}", value)
+	}
+}
+
+impl From<u64> for UpdateName {
+	fn from(value: u64) -> Self {
+		Self(value, UpdateName::left_padded_string(value))
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use crate::chain::chainmonitor::Persist;
+	use crate::chain::ChannelMonitorUpdateStatus;
+	use crate::events::{ClosureReason, MessageSendEventsProvider};
+	use crate::ln::functional_test_utils::*;
+	use crate::util::test_utils::{self, TestLogger, TestStore};
+	use crate::{check_added_monitors, check_closed_broadcast};
+
+	const EXPECTED_UPDATES_PER_PAYMENT: u64 = 5;
+
+	fn default_test_persister<ES: Deref + Clone, SP: Deref + Clone>(
+		read_only: bool,
+		entropy_source: ES,
+		signer_provider: SP,
+	) -> MonitorUpdatingPersister<Box<TestStore>, Box<TestLogger>, ES, SP>
+	where
+		ES::Target: EntropySource + Sized,
+		SP::Target: SignerProvider + Sized,
+	{
+		MonitorUpdatingPersister {
+			kv_store: Box::new(TestStore::new(read_only)),
+			logger: Box::new(TestLogger::new()),
+			maximum_pending_updates: 11,
+			entropy_source,
+			signer_provider,
+		}
+	}
```

Comment:
Yeah fair, at this point it's outgrown its convenience since it takes most of the members as args. Not sure how I ended up with the Box, it might have been from an older type definition.

Reply:
Probably because those need to outlive the lifetime of the method, which isn't possible with local variables.